cpp/src/plasma/CMakeLists.txt

```diff
@@ -128,8 +130,10 @@ if ("${COMPILER_FAMILY}" STREQUAL "gcc")
     " -Wno-conversion")
 endif()
 
+# We use static libraries for the plasma_store_server executable so that it can
+# be copied around and used in different locations.
 add_executable(plasma_store_server store.cc)
-target_link_libraries(plasma_store_server plasma_shared ${PLASMA_LINK_LIBS})
+target_link_libraries(plasma_store_server plasma_static ${PLASMA_STATIC_LINK_LIBS})
```

Comment:
Aside, we need to go through our linking setup and arrange things so that using `plasma_static` will pull in its transitive dependencies. I opened https://issues.apache.org/jira/browse/ARROW-3563